demo_nodes_cpp/src/topics/talker_loaned_message.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/register_node_macro.hpp"
+#include "rcutils/cmdline_parser.h"
+
+#include "std_msgs/msg/string.hpp"
+
+#include "demo_nodes_cpp/visibility_control.h"
+
+using namespace std::chrono_literals;
+
+namespace demo_nodes_cpp
+{
+// Create a Talker class that subclasses the generic rclcpp::Node base class.
+// The main function below will instantiate the class as a ROS node.
+class LoanedMessageTalker : public rclcpp::Node
+{
+public:
+  DEMO_NODES_CPP_PUBLIC
+  explicit LoanedMessageTalker(const rclcpp::NodeOptions & options)
+  : Node("loaned_message_talker", options)
+  {
+    // Create a function for when messages are to be sent.
+    setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+    std::vector<std::string> args = options.arguments();
+    if (find_command_option(args, "-h")) {
+      print_usage();
+      rclcpp::shutdown();
+    } else {
+      std::string tmptopic = get_command_option(args, "-t");
+      if (!tmptopic.empty()) {
+        topic_name_ = tmptopic;
+      }
+      auto publish_message =
+        [this]() -> void
+        {
+          // We loan a message here and don't allocate the memory on the stack.
+          // For middlewares which support message loaning, this means the middleware
+          // completely owns the memory for this message.
+          // This enables a zero-copy message transport for middlewares with shared memory
+          // capabilities.
+          // If the middleware doesn't support this, the loaned message will be allocated
+          // with the allocator instance provided by the publisher.
+          auto loaned_msg = pub_->loan_message();
```

Comment:
Since we are targeting this iteration to POD, does it make sense to use a message that is POD (rather than String).

Reply:
I think it's important to show that non-pod still work transparently. Proposal: We introduce two topics, a `pod_chatter` and a `non_pod_chatter` or something similar.

Reply:
I'm good with that proposal.